google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineRunConfigurationEditor.java

```diff
@@ -158,4 +160,29 @@ public void setAnchor(JComponent anchor) {
     myWebArtifactToDeployLabel.setAnchor(anchor);
     myPortLabel.setAnchor(anchor);
   }
+
+  /**
+   * Initializes the 'artifact to deploy' combobox with the eligible artifact deployment sources.
+   */
+  private void initDeploymentSources() {
+    myArtifactComboBox.setRenderer(
+        new ListCellRendererWrapper<Artifact>() {
+          @Override
+          public void customize(
+              JList list, Artifact value, int index, boolean selected, boolean hasFocus) {
+            if (value != null) {
+              setIcon(value.getArtifactType().getIcon());
+              setText(value.getName());
+            }
+          }
+        });
+
+    List<AppEngineArtifactDeploymentSource> deploymentSources = Lists.newArrayList();
+    deploymentSources.addAll(AppEngineUtil.createArtifactDeploymentSources(myProject));
```

Comment:
So `AppEngineUtil.createArtifactDeploymentSources(myProject)` essentially does the same as previous `collectAppEngineStandardArtifacts` implementation did, but using the standard way, correct?

Reply:
exactly. my goal is to get test coverage here, but that will first require wider refactoring of AppEngineUtil (which I'll do separately)